### DISCUSSION TITLE
Failed to install

### DISCUSSION THREAD

**Discussion by lmanchon** [KarchinLab/open-cravat#301] [Usability & Feature Requests]
--Hi,

i try to install open-cravat with pip on linux with python 3.7.12:

 pip3 install --user open-cravat
Collecting open-cravat
  Downloading open-cravat-2.2.7.tar.gz (3.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 8.4 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Requirement already satisfied: pyyaml in /storage/replicated/cirad/binaries/ifb/miniconda3/envs/perl-5.38.2/lib/python3.7/site-packages (from open-cravat) (6.0)
Collecting requests (from open-cravat)
  Downloading requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
Collecting requests-toolbelt (from open-cravat)
  Downloading requests_toolbelt-1.0.0-py2.py3-none-any.whl.metadata (14 kB)
Collecting pyliftover (from open-cravat)
  Downloading pyliftover-0.4.1-py3-none-any.whl.metadata (3.4 kB)
Collecting websockets (from open-cravat)
  Downloading websockets-11.0.3-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.6 kB)
Collecting markdown (from open-cravat)
  Downloading Markdown-3.4.4-py3-none-any.whl.metadata (6.9 kB)
Collecting aiohttp (from open-cravat)
  Downloading aiohttp-3.8.6-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.7 kB)
Collecting chardet>=3.0.4 (from open-cravat)
  Using cached chardet-5.2.0-py3-none-any.whl.metadata (3.4 kB)
Collecting aiosqlite (from open-cravat)
  Downloading aiosqlite-0.19.0-py3-none-any.whl.metadata (4.3 kB)
Collecting oyaml (from open-cravat)
  Downloading oyaml-1.0-py2.py3-none-any.whl.metadata (1.2 kB)
Collecting intervaltree (from open-cravat)
  Downloading intervaltree-3.1.0.tar.gz (32 kB)
  Preparing metadata (setup.py) ... done
Collecting xlsxwriter (from open-cravat)
  Downloading XlsxWriter-3.2.0-py3-none-any.whl.metadata (2.6 kB)
Collecting openpyxl (from open-cravat)
  Downloading openpyxl-3.1.3-py2.py3-none-any.whl.metadata (2.5 kB)
Collecting twobitreader (from open-cravat)
  Downloading twobitreader-3.1.7.tar.gz (9.2 kB)
  Preparing metadata (setup.py) ... done
Collecting nest-asyncio (from open-cravat)
  Downloading nest_asyncio-1.6.0-py3-none-any.whl.metadata (2.8 kB)
Collecting psutil (from open-cravat)
  Downloading psutil-6.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (21 kB)
Collecting mpmath (from open-cravat)
  Downloading mpmath-1.3.0-py3-none-any.whl.metadata (8.6 kB)
Collecting pyvcf (from open-cravat)
  Downloading PyVCF-0.6.8.tar.gz (34 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in PyVCF setup command: use_2to3 is invalid.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by jasminebro** [comment]
@lmanchon Hi, can you please try to update your python version and try pip install again?

**Comment by lmanchon** [comment]
Hi,

i have updated to python version 3.9.13 and it works now. Thank you.

Which module to install to have a column with the NM reference ?  --> refseq or gencode modules  ?

**Comment by lmanchon** [comment]
refseq module not found:

oc module install refseq
No modules to install found

How to do to install this module ?

**Comment by jasminebro** [comment]
@lmanchon Hi, apologies for the confusion. A RefSeq annotator is currently in the works. I just spoke with our IT team. Expect a release no later than December 2024. We currently have the Ensemble transcript information as an included column in the output. I will leave this issue "open" and respond once the new annotator is released. 

Thank you for your time and using OpenCRAVAT. Please let us know if you need anything else.
